src/common/com/intellij/plugins/haxe/model/HaxeClassModel.java

```diff
@@ -39,6 +42,37 @@ public HaxeClassModel(HaxeClass haxeClass) {
     this.haxeClass = haxeClass;
   }
 
+  public HaxeProjectModel getProject() {
+    return HaxeProjectModel.fromElement(haxeClass);
+  }
+
+  public HaxeModuleModel getModule() {
+    return HaxeModuleModel.fromElement(haxeClass);
+  }
+
+  private HaxeFileModel file = null;
+
+  public HaxeFileModel getFile() {
+    if (file == null) file = new HaxeFileModel((HaxeFile)haxeClass.getContainingFile());
+    return file;
+  }
+
+  @Nullable
+  public FqInfo getFqInfo() {
+    PsiFile file = haxeClass.getContainingFile();
+    String directory = getProject().getPathToDirectory(file.getParent());
+    if (directory == null) return null;
+    String className = haxeClass.getName();
+    String fileName = HaxeFileNameUtils.removeExtension(file.getName());
+
+    return new FqInfo(directory.replace('/', '.'), fileName, className);
+  }
+
+  public HaxePackageModel getPackage() {
```

Comment:
I don't know about the "should" part.
But I have tried to not match the package with the folder in the classPath and haxe compiler can't find it.
As far as I know haxe compiler use directory structure to find types and do not mess with files that are not referenced.
So can't find a Type if it's not in the right folder and with the right `package;`